components/script/dom/htmlformelement.rs

```diff
@@ -395,6 +397,28 @@ impl HTMLFormElementMethods for HTMLFormElement {
             }
         }
 
+        // Step 4
+        let past_names_map = self.past_names_map.borrow();
+        for (key, val) in past_names_map.iter() {
+            let entry = SourcedName {
+                name: key.clone(),
+                element: DomRoot::from_ref(&*val.0),
+                source: SourcedNameSource::Past(now()-val.1), // calculate difference now()-val.1 to find age
+            };
+            sourcedNamesVec.push(entry);
+        }
+
+        // Step 5
+        // TODO need to sort as per spec. This is a partially implemented function.
+        // Kindly guide us on how to refine this function further.
+        sourcedNamesVec.sort_by(|a, b| a.name.partial_cmp(&b.name).unwrap());
+
+        // Step 6
+        sourcedNamesVec.retain(|sn| !sn.name.to_string().is_empty());
+
+        // Step 7
+        // Q1. Unable to clearly understand. It seems to contradict with step 4.
```

Comment:
@jdm Can you kindly explain how to implement step 7? It seems to be contradicting with step 4 in the [spec](https://html.spec.whatwg.org/multipage/forms.html#the-form-element:supported-property-names).

Reply:
The idea here is that the list of names returned in step 8 should not contain any duplicates, and when removing any duplicates the first instance of the name should be retained.

Reply:
It should be enough to add a check in the implementation of step 8 to only push the name if `namesVec.find(elem.name.clone()).is_none()` is true.

Reply:
Done.